comps/imagegen/imagegen.py

```diff
@@ -0,0 +1,77 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import base64
+import os
+import sys
+import time
+from io import BytesIO
+
+import numpy as np
+import tritonclient.http as httpclient
+from PIL import Image
+from tritonclient.utils import *
+
+from comps import Base64ByteStrDoc, ServiceType, TextDoc, opea_microservices, opea_telemetry, register_microservice
+
+
+@opea_telemetry
+def generate_image(*, text, triton_endpoint):
+    start = time.time()
+
+    with httpclient.InferenceServerClient(triton_endpoint, network_timeout=1000 * 300) as client:
+        queries = [text]
+        input_arr = [np.frombuffer(bytes(q, "utf8"), dtype=np.uint8) for q in queries]
+        max_size = max([a.size for a in input_arr])
+        input_arr = [np.pad(a, (0, max_size - a.size)) for a in input_arr]
+        input_arr = np.stack(input_arr)
+
+        inputs = [httpclient.InferInput("INPUT0", input_arr.shape, "UINT8")]
+        inputs[0].set_data_from_numpy(input_arr, binary_data=True)
+
+        outputs = [
+            httpclient.InferRequestedOutput("OUTPUT0"),
+        ]
+
+        ## TODO acwrenn
+        ## Parameterize for other ImageGen models?
+        model_name = "stability"
+        response = client.infer(
+            model_name,
+            inputs,
+            request_id=str(1),
+            outputs=outputs,
+            timeout=1000 * 300,
```

Comment:
Same here. Would it make more sense to set this to?
`timeout = network_timeout`